client/src/lib.rs

```diff
@@ -503,23 +491,34 @@ pub enum ClientError {
     IOError(#[from] std::io::Error),
 }
 
+pub trait FromSigner {
+    fn from_signer(&self) -> &dyn Signer;
+}
```

Comment:
This trait triggers a `clippy` lint in our CI [here](https://github.com/coral-xyz/anchor/actions/runs/9722298001/job/26836009301?pr=3057#step:8:492). Could you maybe change the trait to something like `AsSigner::as_signer` instead?